core/src/test/java/org/apache/iceberg/TestCreateSnapshotEvent.java

```diff
@@ -0,0 +1,120 @@
+    Assert.assertEquals("Table should start empty", 0, listManifestFiles().size());
+
+    table.newAppend().appendFile(FILE_A).commit();
+    Assert.assertNotNull(currentEvent);
+    Assert.assertEquals("Added records in the table should be 1",
+            "1", currentEvent.summary().get("added-records"));
+    Assert.assertEquals("Added files in the table should be 1",
+            "1", currentEvent.summary().get("added-data-files"));
+    Assert.assertEquals("Total records in the table should be 1",
+            "1", currentEvent.summary().get("total-records"));
+    Assert.assertEquals("Total data files in the table should be 1",
+            "1", currentEvent.summary().get("total-data-files"));
+
+    table.newAppend().appendFile(FILE_A).commit();
+    Assert.assertNotNull(currentEvent);
+    Assert.assertEquals("Added records in the table should be 1",
+            "1", currentEvent.summary().get("added-records"));
+    Assert.assertEquals("Added files in the table should be 1",
+            "1", currentEvent.summary().get("added-data-files"));
+    Assert.assertEquals("Total records in the table should be 2",
+            "2", currentEvent.summary().get("total-records"));
+    Assert.assertEquals("Total data files in the table should be 2",
+            "2", currentEvent.summary().get("total-data-files"));
+  }
+
+  @Test
+  public void testAppendAndDeleteCommitEvent() {
+    Assert.assertEquals("Table should start empty", 0, listManifestFiles().size());
+
+    table.newAppend().appendFile(FILE_A).appendFile(FILE_B).commit();
+    Assert.assertNotNull(currentEvent);
+    Assert.assertEquals("Added records in the table should be 2",
+            "2", currentEvent.summary().get("added-records"));
+    Assert.assertEquals("Added files in the table should be 2",
+            "2", currentEvent.summary().get("added-data-files"));
+    Assert.assertEquals("Total records in the table should be 2",
+            "2", currentEvent.summary().get("total-records"));
+    Assert.assertEquals("Total data files in the table should be 2",
+            "2", currentEvent.summary().get("total-data-files"));
+
+    table.newDelete().deleteFile(FILE_A).commit();
+    Assert.assertNotNull(currentEvent);
+    Assert.assertEquals("Deleted records in the table should be 1",
+            "1", currentEvent.summary().get("deleted-records"));
+    Assert.assertEquals("Deleted files in the table should be 1",
+            "1", currentEvent.summary().get("deleted-data-files"));
+    Assert.assertEquals("Total records in the table should be 1",
+            "1", currentEvent.summary().get("total-records"));
+    Assert.assertEquals("Total data files in the table should be 1",
+            "1", currentEvent.summary().get("total-data-files"));
+  }
+
+  class MyListener implements Listener<CreateSnapshotEvent> {
+    @Override
+    public void notify(CreateSnapshotEvent event) {
+      System.out.println("Event :: " + event.summary());
```

Comment:
Please remove all println calls once the PR is ready.

Reply:
this is a miss, removing it.